### DISCUSSION TITLE
How Data format conversion could be perfomred

### DISCUSSION THREAD

**Discussion by pachori91** [edgexfoundry/community#148] [General]
Hi Team,

I am using Edgex 3.0 and able to read sensor data by using device-REST Service and also able to export sensor data by using Http-Export.

My Requirement is to perform data format conversion on data read from the sensor before exporting it to the destination point.

can some one please suggest me for same?

**Comment by lenny-goodell** [comment]
@pachori91 , see here for the built in data format conversions for App Services: https://docs.edgexfoundry.org/3.0/microservices/application/AppServiceConfigurable/#transform
These are available in the `http-export` profile 
These are very simple. 

If you need something else you will need to write a custom App Service with a custom  `conversion` pipeline function.
Here are a few examples:
- https://github.com/edgexfoundry/edgex-examples/tree/v3.0.0/application-services/custom/fledge-export
  - This one is probably most applicable as it does the conversion and then the HTTP export
- https://github.com/edgexfoundry/edgex-examples/tree/v3.0.0/application-services/custom/cloud-event
- https://github.com/edgexfoundry/edgex-examples/tree/v3.0.0/application-services/custom/influxdb-export

Also, see the [app-service-template](https://github.com/edgexfoundry/app-functions-sdk-go/tree/main/app-service-template#readme) which is used to jump start the creation of a new custom App Service with complete Makefile, Dockerfile, etc (examples don't have this).

**Reply by pachori91** [reply]
Thanks @lenny-intel  for your reply,

As my requirment is to perform json to json transformation (Like wise JOLT Transformation).
Further My question is-
Is there any built in service to perform Json to Json Transformation?
I need to write new custom application service for my use case?

**Comment by lenny-goodell** [comment]
> Is there any built in service to perform Json to Json Transformation? I need to write new custom application service for my use case?

Nope, just transform the Event/Reading objects to JSON or XML.
